harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -99,6 +139,406 @@ def test_minimal_vm(api_client, image_id, unique_vm_name, wait_timeout):
+
+    Prerequisite:
+        Setting opensuse-image-url set to a valid URL for
+        an opensuse image.
+
+    Args:
+        resource (tuple): tuple of name(s) & value that can be deconstructed
+        api_client (HarvesterAPI): HarvesterAPI client
+        image_id (str): corresponding image_id from fixture
+        wait_timeout (int): seconds for wait timeout from fixture
+
+    Raises:
+        AssertionError: when vm can not be created, all vms should be allowed to be created
+
+    Steps:
+    1. build vm object specs for outlandish resource(s) under test
+    2. request to build the vm, assert that succeeds
+    3. check for conditions of guest not running and vm being unschedulable
+    4. delete vm and volumes
+
+    Expected Result:
+    - building vm with outlandish resource requests to be successful
+    - asserting that the status condition of the vm that is built to not be running
+    - asserting that the status condition of the vm that is built to be unschedulable
+    - assert deleting vm and volumes to be successful
+    """
+    unique_name_for_vm = _adjusted_unique_name()
+    resource_name, resource_val = resource
+    overall_vm_obj = {
+        'cpu': 1,
+        'mem': 2,
+        'disk': 10
+    }
+
+    if resource_name == 'cpu':
+        overall_vm_obj['cpu'] = resource_val
+    if resource_name == 'mem':
+        overall_vm_obj['mem'] = resource_val
+    if resource_name == 'disk':
+        overall_vm_obj['disk'] = resource_val
+    if resource_name == 'mem_and_cpu':
+        overall_vm_obj["mem"] = resource_val
+        overall_vm_obj["cpu"] = resource_val
+    if resource_name == 'mem_cpu_and_disk':
+        overall_vm_obj["mem"] = resource_val
+        overall_vm_obj["cpu"] = resource_val
+        overall_vm_obj["disk"] = resource_val
+
+    vm = api_client.vms.Spec(overall_vm_obj['cpu'], overall_vm_obj['mem'])
+    vm.add_image("disk-0", image_id, size=overall_vm_obj.get('disk'))
+    code, data = api_client.vms.create(unique_name_for_vm, vm)
+    assert 201 == code, (code, data)
+
+    endtime = datetime.now() + timedelta(seconds=wait_timeout)
+    while endtime > datetime.now():
+        code, data = api_client.vms.get_status(unique_name_for_vm)
+        if 200 == code and len(data.get('status', {}).get('conditions', [])) > 1:
+            check_dict = {
+                'not_running': None,
+                'unschedulable': None
+            }
+            for status_condition in data.get('status', {}).get('conditions', []):
+                if status_condition.get('reason') == 'GuestNotRunning':
+                    check_dict['not_running'] = True
+                if status_condition.get('reason') == 'Unschedulable':
+                    check_dict['unschedulable'] = True
+            assert check_dict["not_running"] is True, "the vm shouldn't be running"
+            assert check_dict["unschedulable"] is True, "the vm shouldn't be schedulable"
```

Comment:
we can just check that the vmi is entering 'Unschedulable' condition as:
```python
endtime = datetime.now() + timedelta(seconds=wait_timeout)
while endtime > datetime.now():
    code, data = api_client.vms.get_status(unique_name_for_vm)
    conds = data['status'].get('conditions', [{}])
    if conds[-1].get('reason') == 'Unschedulable':
        break
    sleep(3)
else:
    raise AssertionError(
        f"VM {unique_name_for_vm} entered unexpected condition\n"
        f"Phase: {data['status'].get('phase')}\n"
        f"VM resources: {overall_vm_obj}\n"
        f"conditions: {conds}\n"
        f"API Status({code}): {data}"
)
```
and delete vm should not be included in the same while loop.